src/test/java/com/puppycrawl/tools/checkstyle/utils/CommonUtilTest.java

```diff
@@ -440,6 +452,42 @@ public void testIsIntNull() {
             CommonUtil.isInt(null));
     }
 
+    @Test
+    public void testGetUriByFilenameFindsAbsoluteResourceOnClasspath() throws Exception {
+        final String filename =
+            "/" + getPackageLocation() + "/InputCommonUtilTest_empty_checks.xml";
+        final URI uri = CommonUtil.getUriByFilename(filename);
+        assertThat("URI is null for: " + filename, uri, is(not(nullValue())));
+    }
+
+    @Test
+    public void testGetUriByFilenameFindsRelativeResourceOnClasspath() throws Exception {
+        final String filename =
+            getPackageLocation() + "/InputCommonUtilTest_empty_checks.xml";
+        final URI uri = CommonUtil.getUriByFilename(filename);
+        assertThat("URI is null for: " + filename, uri, is(not(nullValue())));
+    }
+
+    @Test
+    public void testGetUriByFilenameFindsResourceRelativeToRootClasspath() throws Exception {
+        final String filename = "commonutil/InputCommonUtilTest_resource.txt";
+        final URI uri = CommonUtil.getUriByFilename(filename);
+        assertThat("URI is null for: " + filename, uri, is(not(nullValue())));
+        assertThat("URI contains com/puppycrawl/tools/checkstyle/utils",
+            uri.toString(), not(containsString("com/puppycrawl/tools/checkstyle/utils")));
```

Comment:
Why aren't we validating the path of the URI. Current assetions just say it shouldn't be something and don't specify what they should be.

Reply:
That's because I want my test assertions to be a representation of _exactly_ what I'm trying to test, nothing more, nothing less. Doing an exact validation on the URI is of course possible, but it's overly strict and it would be less good a reflection of what I'm actually trying to illustrate and test.

Reply:
@romani What do you think of this?

Reply:
I am ok with extra validation. 
But please move `com/puppycrawl/tools/checkstyle/utils` to variable.